core-rust/state-manager/src/mempool/pending_transaction_result_cache.rs

```diff
@@ -451,17 +451,13 @@ impl PendingTransactionResultCache {
     pub fn track_committed_transactions(
         &mut self,
         current_timestamp: SystemTime,
-        previous_state_version: StateVersion,
-        hashes: Vec<IntentHash>,
+        hashes: Vec<(StateVersion, IntentHash)>,
     ) {
-        let mut resultant_state_version = previous_state_version;
-        for intent_hash in hashes {
+        for (state_version, intent_hash) in hashes {
```

Comment:
:warning: (_important note from author_):

I believe that this is a bugfix:
Before this PR, this method might have been populating wrong `AtState::Committed.state_version` (i.e. it was computing `base_state_version + index`, but the `hashes` only contained user transactions [`intent_hash_if_user()`, remember?], which means that "round change" transactions were disrupting the computed versions).
I switched to passing explicit, definitely-right state version for every intent hash.

Reply:
Yup - agreed. This is much better 👍